floss/main.py

```diff
@@ -222,31 +314,7 @@ def set_logging_level(should_debug=False, should_verbose=False):
         logging.basicConfig(level=logging.INFO)
     else:
         logging.basicConfig(level=logging.WARNING)
-
-        # ignore messages like:
-        # WARNING:EmulatorDriver:error during emulation of function: BreakpointHit at 0x1001fbfb
-        # ERROR:EmulatorDriver:error during emulation of function ... DivideByZero: DivideByZero at 0x10004940
-        # TODO: probably should modify emulator driver to de-prioritize this
-        logging.getLogger("EmulatorDriver").setLevel(logging.CRITICAL)
-
-        # ignore messages like:
-        # WARNING:Monitor:logAnomaly: anomaly: BreakpointHit at 0x1001fbfb
-        logging.getLogger("Monitor").setLevel(logging.ERROR)
-
-        # ignore messages like:
-        # WARNING:envi/codeflow.addCodeFlow:parseOpcode error at 0x1001044c: InvalidInstruction("'660f3a0fd90c660f7f1f660f6fe0660f' at 0x1001044c",)
-        logging.getLogger("envi/codeflow.addCodeFlow").setLevel(logging.ERROR)
-
-
-        # ignore messages like:
-        # WARNING:vtrace.platforms.win32:LoadLibrary C:\Users\USERNA~1\AppData\Local\Temp\_MEI21~1\vtrace\platforms\windll\amd64\symsrv.dll: [Error 126] The specified module could not be found
-        # WARNING:vtrace.platforms.win32:LoadLibrary C:\Users\USERNA~1\AppData\Local\Temp\_MEI21~1\vtrace\platforms\windll\amd64\dbghelp.dll: [Error 126] The specified module could not be found
-        logging.getLogger('vtrace.platforms.win32').setLevel(logging.ERROR)
-
-        # ignore messages like:
-        # WARNING:plugins.arithmetic_plugin.XORPlugin:identify: Invalid instruction encountered in basic block, skipping: 0x4a0637
-        logging.getLogger("floss.plugins.arithmetic_plugin.XORPlugin").setLevel(logging.ERROR)
-        logging.getLogger("floss.plugins.arithmetic_plugin.ShiftPlugin").setLevel(logging.ERROR)
+        set_log_config(should_debug, should_verbose)
```

Comment:
The only reason I didn't call set_log_config() from every case is because I figured logging.BasicConfig() would do the same thing, and it'd be redundant. set_log_config() is called only if logging.BasicConfig() sets the loggers to the WARNING level, because set_log_config() sets most of the loggers  to ERROR level, and what's not explicitly set is already set by logging.BasicConfig(). Calling set_log_config() when logging.BasicConfig() has already set the loggers to DEBUG or INFO level would just accomplish the same thing twice it'd think.

Reply:
I think that this routine should be called from all cases (or de-dented) for symmetry. Maybe there is some duplication, but I believe this is ok, since otherwise, its confusing why the final case is special.

FYI, this type of reasoning is the perfect type of explanation to go into code comments.